components/widgets/settings/bool_setting.jsx

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Setting from './setting.jsx';
+
+export default class BoolSetting extends React.Component {
+    static propTypes = {
+        id: PropTypes.string.isRequired,
+        label: PropTypes.node.isRequired,
+        labelClassName: PropTypes.string,
+        helpText: PropTypes.node,
+        placeholder: PropTypes.string.isRequired,
+        value: PropTypes.bool.isRequired,
+        inputClassName: PropTypes.string,
+        onChange: PropTypes.func,
+    };
+
+    static defaultProps = {
+        labelClassName: '',
+        inputClassName: '',
+    };
+
+    handleChange = (e) => {
+        this.props.onChange(this.props.id, e.target.checked);
+    }
+
+    render() {
+        return (
+            <Setting
+                label={this.props.label}
+                labelClassName={this.props.labelClassName}
+                inputClassName={this.props.inputClassName}
+                helpText={this.props.helpText}
+                inputId={this.props.id}
+            >
+                <div className='checkbox'>
+                    <label>
+                        <input
+                            id={this.props.id}
+                            type='checkbox'
+                            checked={this.props.value}
+                            onChange={this.handleChange}
+                        />
+                        <span>{this.props.placeholder}</span>
```

Comment:
Do we ready want _three_ ways of showing some text alongside the checkbox. Currently we have:
- `display_name`: Which is shown bold above the checkbox
- `placeholder`: Which is shown on the right side of the checkbox
- `help_text`: Which is shown below the checkbox

![Screenshot from 2019-07-16 14-08-21](https://user-images.githubusercontent.com/16541325/61293079-2f98ec80-a7d3-11e9-9743-8cdd7f16c60d.png)

My proposal would be to get rid of the `placeholder` field and move `display_name` to where `placeholder` currently is. This would roughly look like this:

![Screenshot from 2019-07-16 14-13-45](https://user-images.githubusercontent.com/16541325/61293419-f01ed000-a7d3-11e9-98e2-c2627283d49a.png)

@hvhallmann @asaadmahmood @aaronrothschild Please let me know what you think about this

Reply:
I'm 2/5 at keeping 3 locations.

I assume the additional text locations are intended to make a selection of a checkbox clearer for the user.

Since we can't fully estimate some of the complex configuration questions we may need to present to the user, I'm personally in favor of having the three pieces of text to err on the side of letting a developer craft a UI that very clearly articulates to the user what they are (about to) do.  Some settings can be scary for a user, because they can potentially cause weird behavior,  perform badly, etc. Any extra contextual information that can be presented to a user is good in that case IMO.

Reply:
I prefer to keep those 3 locations, first because it follows the UI pattern than further components, keeping the users with the same look and feel.  And I agree with aaron also, because it let users with more options to craft their UI.